servlet/src/main/java/com/linecorp/armeria/server/servlet/DefaultServletContext.java

```diff
@@ -0,0 +1,573 @@
+    }
+
+    @Override
+    public Map<String, ServletRegistration> getServletRegistrations() {
+        return servletRegistrationMap;
+    }
+
+    @Override
+    public FilterRegistration addFilter(String filterName, String className) {
+        try {
+            requireNonNull(filterName, "filterName");
+            requireNonNull(className, "className");
+            return addFilter(filterName, (Class<Filter>) Class.forName(className));
+        } catch (ClassNotFoundException e) {
+            logger.error("Add filter failed: ", e);
+        }
+        return null;
+    }
+
+    @Override
+    public FilterRegistration addFilter(String filterName, Filter filter) {
+        requireNonNull(filterName, "filterName");
+        requireNonNull(filter, "filter");
+        checkArgument(!isInitialized(), "Can't add filter after server is initialized");
+        final FilterRegistration registration =
+                new FilterRegistration(filterName, filter, this, filterUrlMapper);
+        filterRegistrationMap.put(filterName,registration);
+        return registration;
+    }
+
+    @Override
+    public FilterRegistration addFilter(String filterName, Class<? extends Filter> filterClass) {
+        requireNonNull(filterName, "filterName");
+        requireNonNull(filterClass, "filterClass");
+        try {
+            return addFilter(filterName, createFilter(filterClass));
+        } catch (ServletException e) {
+            logger.error("Add filter failed: ", e);
+        }
+        return null;
+    }
+
+    @Override
+    public <T extends Filter> T createFilter(Class<T> clazz) throws ServletException {
+        requireNonNull(clazz, "clazz");
+        try {
+            return clazz.newInstance();
+        } catch (InstantiationException | IllegalAccessException e) {
+            logger.error("Create filter failed: ", e);
+        }
+        return null;
+    }
+
+    @Override
+    public javax.servlet.FilterRegistration getFilterRegistration(String filterName) {
+        requireNonNull(filterName, "filterName");
+        return filterRegistrationMap.get(filterName);
+    }
+
+    @Override
+    public Map<String, FilterRegistration> getFilterRegistrations() {
+        return filterRegistrationMap;
+    }
+
+    @Override
+    public SessionCookieConfig getSessionCookieConfig() {
+        return null;
+    }
+
+    @Override
+    public void setSessionTrackingModes(Set<SessionTrackingMode> sessionTrackingModes) {
+        requireNonNull(sessionTrackingModes, "sessionTrackingModes");
+        checkArgument(!isInitialized(), "Can't set session tracking mode after server is initialized");
+        sessionTrackingModeSet = sessionTrackingModes;
+    }
+
+    @Override
+    public Set<SessionTrackingMode> getDefaultSessionTrackingModes() {
+        return defaultSessionTrackingModeSet;
+    }
+
+    @Override
+    public Set<SessionTrackingMode> getEffectiveSessionTrackingModes() {
+        if (sessionTrackingModeSet == null) {
```

Comment:
We can just return `sessionTrackingModeSet` if we do
```java
private Set<SessionTrackingMode> sessionTrackingModeSet = defaultSessionTrackingModeSet;
```
Also we can do:
```java
private static final Set<SessionTrackingMode> defaultSessionTrackingModeSet =
        Sets.immutableEnumSet(SessionTrackingMode.COOKIE, SessionTrackingMode.URL);
```

Reply:
When `defaultSessionTrackingModeSet` is immutable and `sessionTrackingModeSet = defaultSessionTrackingModeSet`, I think `sessionTrackingModeSet` has never changed. We should keep `defaultSessionTrackingModeSet` is not immutable and convert `sessionTrackingModeSet` to immutable in `pack()` method.

Reply:
> When defaultSessionTrackingModeSet is immutable and sessionTrackingModeSet = defaultSessionTrackingModeSet, I think sessionTrackingModeSet has never changed. 

I mean we can set `defaultSessionTrackingModeSet` to `sessionTrackingModeSet` when the class is initialized:
```java
public class DefaultServletContext {
    private static final Set<SessionTrackingMode> defaultSessionTrackingModeSet =
        Sets.immutableEnumSet(SessionTrackingMode.COOKIE, SessionTrackingMode.URL);

    private Set<SessionTrackingMode> sessionTrackingModeSet = defaultSessionTrackingModeSet;

    ...

    // A user can set sessionTrackingModeSet using the setter.
    @Override
    public void setSessionTrackingModes(Set<SessionTrackingMode> sessionTrackingModes) {
        requireNonNull(sessionTrackingModes, "sessionTrackingModes");
        ...
    }
}
```
Sorry about the misleading comment. 😉

Reply:
Yes, I understood. I want to confirm do you mean [this comment #2686](https://github.com/line/armeria/pull/2686#discussion_r429121931) (comment) exclude setSessionTrackingModes()?